applications/asset_tracker_v2/CMakeLists.txt

```diff
@@ -39,3 +39,8 @@ if (CONFIG_BOARD_QEMU_X86 OR CONFIG_BOARD_NATIVE_POSIX)
         target_compile_options(app PRIVATE
 	        -DCONFIG_LTE_NEIGHBOR_CELLS_MAX=10)
 endif()
+
+# Enable Link Time Optimization for the application code
+if(CMAKE_C_COMPILER_ID STREQUAL "GNU" AND CONFIG_SIZE_OPTIMIZATIONS)
```

Comment:
I think that in order to easily disable lto on compilation units, in case users notice unexpected behavior, then create a app specific setting here: https://github.com/nrfconnect/sdk-nrf/blob/main/applications/asset_tracker_v2/Kconfig

in the form of (feel free to adjust name / location, etc.:
```
config ATV_V2_LTO_OPTIMIZATION
        bool "Compile app library with lto"
        default y if SIZE_OPTIMIZATIONS
        help
          Compile source code in app library with lto.
```